tools/run.py

```diff
@@ -0,0 +1,392 @@
+    nth_mixture['files'] = nth_mixture['files'][sort_idx]
+    nth_mixture['class'] = nth_mixture['class'][sort_idx]
+    nth_mixture['event_onoffsets'] = nth_mixture['event_onoffsets'][sort_idx]
+    #nth_mixture['sample_onoffsets'] = nth_mixture['sample_onoffsets'][sort_idx]
+    nth_mixture['trajectory'] = nth_mixture['trajectory'][sort_idx]
+    nth_mixture['isMoving'] = nth_mixture['isMoving'][sort_idx]
+    nth_mixture['isFlippedMoving'] = nth_mixture['isFlippedMoving'][sort_idx]
+    nth_mixture['speed'] = nth_mixture['speed'][sort_idx]
+    nth_mixture['rirs'] = np.array(nth_mixture['rirs'],dtype=object)
+    nth_mixture['rirs'] = nth_mixture['rirs'][sort_idx]
+    new_doas = np.zeros(len(sort_idx),dtype=object)
+    new_sample_onoffsets = np.zeros(len(sort_idx),dtype=object)
+    upd_idx = 0
+    for idx in sort_idx:
+        new_doas[upd_idx] = nth_mixture['doa_azel'][idx].T
+        new_sample_onoffsets[upd_idx] = nth_mixture['sample_onoffsets'][idx]
+        upd_idx += 1
+    nth_mixture['doa_azel'] = new_doas
+    nth_mixture['sample_onoffsets'] = new_sample_onoffsets 
+
+    return nth_mixture
+
+def main():
+    '''
+    main function to trigger the data generation
+    '''
+
+    # parse config arguments
+    args = parse_args()
+    cfg = load_config(args, args.path_to_config)
+    print('generating data using parameters:')
+    print(yaml.dump(dict(cfg), allow_unicode=True, default_flow_style=False))
+
+    
+    event_filenames = get_sound_event_filenames(cfg.PATH_TO_SOUND_EVENT_FILES)
+    event_durs = get_sound_event_filedurs(cfg.PATH_TO_SOUND_EVENT_DURS)
+
+
+    fold_names = cfg.FOLD_NAMES
+    snr_range = cfg.SNR_RANGE
+    max_polyphony = cfg.MAX_POLYPHONY
+    mixtures = []
+
+    # iterate over fold names
+    for ifold, fold in enumerate(fold_names):
+        
+        fold_event_filenames, fold_event_durs = get_fold_files_and_durs(
+                fold, 
+                event_filenames, 
+                event_durs)
+        n_mixtures_per_fold = cfg.N_MIX_PER_FOLD[ifold]
+
+        # iterate over rooms
+        fold_rooms = cfg.FOLD_ROOMS[fold]
+        room_mixtures=[]
+        for room_name in fold_rooms:
+            fold_mixture = {'mixture': []}
+            fold_mixture['roomidx'] = room_name
+
+            # get the room's relevant info
+            path_to_room_files = get_path_to_room_files(room_name)
+            room_trajs = get_room_trajectories(path_to_room_files)
+            n_traj = len(room_trajs)
+            traj_doas = get_traj_doas(room_trajs, n_traj)
+            sample_counter = 0
+
+            # generate each mixture
+            for nmix in range(n_mixtures_per_fold):
+                print('Room {}, generating mixture {}'.format(room_name, nmix+1))
+
+                nth_mixture = initialize_mixture_metadata(room_name, snr_range)
+
+                nth_mixture = populate_mixture(cfg, sample_counter, fold_event_durs, fold_event_filenames, n_traj, traj_doas, max_polyphony, nth_mixture)
```

Comment:
these two functions are super intricate. I will focus on further breaking these down and proposing API functionalities from them. I'm thinking that the audio synthesis could perhaps be integrated here.